packages/rs-dpp/src/state_transition/state_transition_factory.rs

```diff
@@ -19,15 +26,103 @@ use crate::{
     prelude::Identifier,
     state_repository::StateRepositoryLike,
     util::json_value::JsonValueExt,
-    ProtocolError,
+    validation::AsyncDataValidatorWithContext,
+    BlsModule, ProtocolError,
 };
+use serde_json::Number;
 use serde_json::Value as JsonValue;
 
 use super::{
-    state_transition_execution_context::StateTransitionExecutionContext, StateTransition,
-    StateTransitionType,
+    state_transition_execution_context::StateTransitionExecutionContext,
+    validation::{
+        validate_state_transition_basic::StateTransitionBasicValidator,
+        validate_state_transition_by_type::StateTransitionByTypeValidator,
+    },
+    StateTransition, StateTransitionType,
 };
 
+#[derive(Default)]
+pub struct StateTransitionFactoryOptions {
+    pub skip_validation: bool,
+}
+
+pub struct StateTransitionFactory<SR, BLS>
+where
+    SR: StateRepositoryLike,
+    BLS: BlsModule,
+{
+    state_repository: Arc<SR>,
+    basic_validator: StateTransitionBasicValidator<SR, StateTransitionByTypeValidator<SR, BLS>>,
+}
+
+impl<SR, BLS> StateTransitionFactory<SR, BLS>
+where
+    SR: StateRepositoryLike,
+    BLS: BlsModule,
+{
+    pub fn new(
+        state_repository: Arc<SR>,
+        basic_validator: StateTransitionBasicValidator<SR, StateTransitionByTypeValidator<SR, BLS>>,
+    ) -> Self {
+        StateTransitionFactory {
+            state_repository,
+            basic_validator,
+        }
+    }
+
+    pub async fn create_from_object(
+        &self,
+        raw_state_transition: JsonValue,
+        options: Option<StateTransitionFactoryOptions>,
+    ) -> Result<StateTransition, ProtocolError> {
+        let options = options.unwrap_or_default();
+
+        if !options.skip_validation {
+            let execution_context = StateTransitionExecutionContext::default();
```

Comment:
Do we always want the context to be the default context here?

Reply:
I did as in other factories we have, so I'm not sure myself

Reply:
The original implementation does indeed have default context, thank you!